### PR TITLE
ATDM: cee-rhel6: swtich from intel-18.0.2_mpich2-3.2 to intel-19.0.3_mpich2-3.2 (ATDV-390)

### PR DESCRIPTION
**Pull request by bartlettroscoe**
This is WIP topic-branch to fix the clang-9.0.1 and intel-19.0.3_mpich2-3.2 cee-rhel6 builds that support the matching SPARC Trilinos Integration builds (see [ATDV-390](https://sems-atlassian-srn.sandia.gov/browse/ATDV-390)).

## How was this tested?

On the CEE RHEL7 machine 'cee-build015' 

```
$ ssh cee-build015

$ cd /scratch/rabartl/Trilinos.base/BUILDS/ATDM/CEE-RHEL6/CTEST_S/

$ ./ctest-s-local-test-driver-cee-rhel6.sh \
  cee-rhel6_clang-9.0.1_openmpi-4.0.3_serial_static_opt \
  cee-rhel6_intel-19.0.3_mpich2-3.2_openmp_static_opt

***
*** /scratch/rabartl/Trilinos.base/Trilinos/cmake/std/atdm/ctest-s-local-test-driver.sh
***

ATDM_TRILINOS_DIR = '/scratch/rabartl/Trilinos.base/Trilinos'

Load some env to get python, cmake, etc ...

Hostname 'cee-build015' matches known ATDM host 'cee-build015' and system 'cee-rhel6'
Setting compiler and build options for build-name 'cee-rhel6-default'
Using CEE RHEL6 compiler stack CLANG-9.0.1_OPENMPI-4.0.3 to build DEBUG code with Kokkos node type SERIAL

Running builds:
    cee-rhel6_clang-9.0.1_openmpi-4.0.3_serial_static_opt
    cee-rhel6_intel-19.0.3_mpich2-3.2_openmp_static_opt

Fri Sep  4 13:35:51 MDT 2020

Running Jenkins driver Trilinos-atdm-cee-rhel6_clang-9.0.1_openmpi-4.0.3_serial_static_opt.sh ...

    Creating directory: Trilinos-atdm-cee-rhel6_clang-9.0.1_openmpi-4.0.3_serial_static_opt

    Creating directory: SRC_AND_BUILD

    See log file Trilinos-atdm-cee-rhel6_clang-9.0.1_openmpi-4.0.3_serial_static_opt/smart-jenkins-driver.out

real    92m15.246s
user    1071m48.970s
sys     115m40.628s

99% tests passed, 2 tests failed out of 2207

Fri Sep  4 15:08:06 MDT 2020

Running Jenkins driver Trilinos-atdm-cee-rhel6_intel-19.0.3_mpich2-3.2_openmp_static_opt.sh ...

    Creating directory: Trilinos-atdm-cee-rhel6_intel-19.0.3_mpich2-3.2_openmp_static_opt

    Creating directory: SRC_AND_BUILD

    See log file Trilinos-atdm-cee-rhel6_intel-19.0.3_mpich2-3.2_openmp_static_opt/smart-jenkins-driver.out

real    206m47.823s
user    2787m55.478s
sys     322m59.914s

99% tests passed, 1 tests failed out of 2328

Fri Sep  4 18:34:54 MDT 2020

Done running all of the builds!
```

That posted to CDash at:

* https://testing.sandia.gov/cdash/index.php?project=Trilinos&date=2020-09-04&filtercount=3&showfilters=1&filtercombine=and&field1=groupname&compare1=61&value1=Experimental&field2=buildname&compare2=65&value2=Trilinos-atdm-&field3=site&compare3=61&value3=cee-build015

showing:

| Site | Build Name | Conf Err | Conf Warn | Conf Test Time | Build Err | Build Warn | Build Test Time | Test Not Run | Test Fail | Test Pass | Test Time | Test Proc Time | Start Test Time | Labels |
| -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- |
cee-build015 | Trilinos-atdm-cee-rhel6_intel-19.0.3_mpich2-3.2_openmp_static_opt-exp | 0 | 3 | 3m 26s | 0 | 50 | 3h 13m 25s | 0 | 1 | 2327 | 9m 29s | 1h 3m 55s | 4 hours ago | (30 labels)
cee-build015 | Trilinos-atdm-cee-rhel6_clang-9.0.1_openmpi-4.0.3_serial_static_opt-exp | 0 | 3 | 2m 35s | 0 | 50 | 1h 8m 55s | 0 | 2 | 2205 | 20m 16s | 2h 33m 31s | 6 hours ago | (30 labels)

Ironically, these builds only works on CEE RHEL7 machines, not CEE RHEL6 machines :-(

### CODE REVIEW

**Comment by bartlettroscoe** [comment]
@e10harvey and @jmgate, this is ready to be merged to 'develop'.  But no huge hurry since I already merged to the 'atdm-nightly-manual-updates' branch in the commit https://github.com/trilinos/Trilinos/commit/a19ea88eee3578c7604b178f4718250eaf80312d so this will run tomorrow.

File: cmake/std/atdm/cee-rhel6/custom_builds.sh [hunk 1]
```diff
@@ -31,12 +31,10 @@ elif atdm_match_any_buildname_keyword \
   # matching of defaults
 
 elif atdm_match_any_buildname_keyword \
-    intel-18.0.2-mpich2-3.2 \
-    intel-18.0.2_mpich2-3.2 \
-    intel-18.0.2 \
-    intel-18 \
```
Comment (jmgate) [inline cmake/std/atdm/cee-rhel6/custom_builds.sh:37]: No
```
intel-19.0.3 \
intel-19 \
```

?

**Review by jmgate — APPROVED** [review]

File: cmake/std/atdm/cee-rhel6/custom_builds.sh (hunk 1 above)
Comment (jmgate) [reply cmake/std/atdm/cee-rhel6/custom_builds.sh:37]: Removing automerge in case this comment requires a change.

Comment (bartlettroscoe) [reply cmake/std/atdm/cee-rhel6/custom_builds.sh:37]: @jmgate, the default 'intel-19.0.3' and 'intel-19' maps to `INTEL-19.0.3_INTELMPI-2018.4` and **not** `ATDM_CONFIG_COMPILER=INTEL-19.0.3_MPICH2-3.2`.  See:

* https://github.com/trilinos/Trilinos/blob/afa21bdbacb31dfd7299d1c5db8be33fb9f98dc5/cmake/std/atdm/cee-rhel6/custom_builds.sh#L39
* https://github.com/trilinos/Trilinos/blob/afa21bdbacb31dfd7299d1c5db8be33fb9f98dc5/cmake/std/atdm/cee-rhel6/custom_builds_unit_tests.sh#L139
* https://github.com/trilinos/Trilinos/blob/afa21bdbacb31dfd7299d1c5db8be33fb9f98dc5/cmake/std/atdm/cee-rhel6/custom_builds_unit_tests.sh#L143

Comment (bartlettroscoe) [reply cmake/std/atdm/cee-rhel6/custom_builds.sh:37]: @jmgate, I think the units in:

* https://github.com/trilinos/Trilinos/blob/afa21bdbacb31dfd7299d1c5db8be33fb9f98dc5/cmake/std/atdm/cee-rhel6/custom_builds_unit_tests.sh

show that this logic matches expected/desired behavior.

**Comment by bartlettroscoe** [comment]
@jmgate, I had this against the 'master' branch against 'develop' on accident.

**Review by jmgate — APPROVED** [review]

**Pull request merged**